src/ucs/arch/atomic.h

```diff
@@ -138,4 +139,59 @@ UCS_DEFINE_ATOMIC_BOOL_CSWAP(16, w);
 UCS_DEFINE_ATOMIC_BOOL_CSWAP(32, l);
 UCS_DEFINE_ATOMIC_BOOL_CSWAP(64, q);
 
+
+#define UCS_ATOMIC_WEAK         UCS_BIT(0)
+#define UCS_ATOMIC_FENCE_LOCK   UCS_BIT(1)
+#define UCS_ATOMIC_FENCE_UNLOCK UCS_BIT(2)
+
+
+static UCS_F_ALWAYS_INLINE int ucs_atomic_memorder(unsigned flags)
+{
+    if (flags & UCS_ATOMIC_FENCE_LOCK) {
+        return __ATOMIC_ACQUIRE;
+    }
+
+    if (flags & UCS_ATOMIC_FENCE_UNLOCK) {
+        return __ATOMIC_RELEASE;
+    }
+
+    return __ATOMIC_RELAXED;
```

Comment:
why are UCS_ATOMIC_xx defined as flags if there is no meaning to combine more than one of them?
also, how does it relate to UCS_DEFINE_ATOMIC_xx macros? should we remove/extend them?